packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/instruction/location.ts

```diff
@@ -0,0 +1,52 @@
+import { PublicKey, TransactionInstruction } from '@solana/web3.js'
+import { NextFunction, Request, Response } from 'express'
+
+import { getRequestIpData } from '../../utils/ipData'
+
+const MEMO_V2_PROGRAM_ID = 'MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr'
+const GEO_MEMO_STRING = 'geo'
+
+/**
+ * Returns a memo instruction containing geo location data.
+ */
+export const location = async (
+  req: Request<
+    unknown,
+    unknown,
+    unknown,
+    {
+      signer?: string
+    }
+  >,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const signer = req.query.signer
+    if (!signer) {
+      res.status(400).send({ error: 'Missing signer parameter' })
+      next()
+    }
+    const location = await getRequestIpData(res.locals.logger, req)
+    const instruction = new TransactionInstruction({
+      keys: signer
```

Comment:
I like this as an optional parameter. If the caller wants to sign the memo for extra verification, they can provide a signer here and then sign the request afterwards. If they exclude it, also fine. 👍